sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala

```diff
@@ -1196,6 +1196,24 @@ class ParquetIOSuite extends QueryTest with ParquetTest with SharedSparkSession
       }
     }
   }
+
+  test("SPARK-34167: read LongDecimals with precision < 10, VectorizedReader off") {
+    readParquetFile(testFile("test-data/decimal32-written-as-64-bit.snappy.parquet"), false) {
```

Comment:
it might be nice to add a comment about how the parquet file was generated. At least state it wasn't from spark but a 3rd party library and it has X in it that is different.